src/lib.rs

```diff
@@ -126,15 +126,49 @@ pub trait Renderable: Send + Sync {
     fn render(&self, context: &mut Context) -> Result<Option<String>>;
 }
 
+pub type TemplateLocation = Path;
+pub trait FileRepository {
```

Comment:
It seems there is a difference between public Liquid and Liquid for Shopify's customers.  I'm trying to better understand it.  Not sure how we should express the difference in our liquid implementation.

These different type of template load systems seem to use different directories (from my quick glance at the `section` documentation).  I wonder if your new trait should take two parameters, a type and a name.  Not sure if the type should be a string or an enum.